src/sema/variables.rs

```diff
@@ -460,6 +461,143 @@ pub fn variable_decl<'a>(
                 ));
             }
 
+            let body;
+            let returns;
```

Comment:
You don't have to declare the variable like this. In Rust you can do `let (body, returns) = if let Type::Struct(str) = ..`